integration-tests/test_client_options.py

```diff
@@ -17,3 +21,25 @@ def test_set_ansible_host_info(insights_client):
     ret = insights_client.run("--ansible-host=foo.example.com", check=False)
     assert "Could not update Ansible hostname" not in ret.stdout
     assert ret.returncode == 0
+
+
+def test_no_upload(insights_client):
+    """Verify no traceback is returned when running '--no-upload' command
+    and the archive is created and saved on the system
+    """
+    archive_saved = "Archive saved at"
+    upload_message = "Successfully uploaded report"
+    archive_directory = "/var/cache/insights-client"
+
+    insights_client.register()
+
+    if os.listdir(archive_directory):
+        shutil.rmtree(archive_directory)
+        os.makedirs(archive_directory)
```

Comment:
hmm removing files here IMHO is not a good idea, and I'd rather work regardless of the content of that directory; also, that directory is statically shipped by the RPM package, so there is no need to manually create it

what I'd do instead is the following:
- before the `--no-upload` invocation, `glob()` to get the content of the cache directory
- after `--no-upload`, `glob()` again to get the new content in the directory
- diff the "after" with the "before", and check that there is at least (or exactly) one element more; tip: convert the list to sets using `set()` and diff them easily

Reply:
done